Tribler/community/market/core/tick.py

```diff
@@ -0,0 +1,325 @@
+        self._public_key = public_key
+        self._signature = signature
+
+    @classmethod
+    def from_database(cls, data):
+        tick_cls = Ask if data[9] else Bid
+        message_id = MessageId(TraderId(str(data[0])), MessageNumber(str(data[1])))
+        order_id = OrderId(TraderId(str(data[0])), OrderNumber(data[2]))
+        return tick_cls(message_id, order_id, Price(data[3], str(data[4])), Quantity(data[5], str(data[6])),
+                        Timeout(data[7]), Timestamp(data[8]), str(data[10].decode('hex')), str(data[11].decode('hex')))
+
+    def to_database(self):
+        return (unicode(self.message_id.trader_id), unicode(self.message_id.message_number),
+                int(self.order_id.order_number), float(self.price), unicode(self.price.wallet_id), float(self.quantity),
+                unicode(self.quantity.wallet_id), float(self.timeout), float(self.timestamp), self.is_ask(),
+                unicode(self._public_key.encode('hex')), unicode(self._signature.encode('hex')))
+
+    @classmethod
+    def from_order(cls, order, message_id):
+        """
+        Create a tick from an order
+
+        :param order: The order that this tick represents
+        :param message_id: The message id for the tick
+        :return: The created tick
+        :rtype: Tick
+        """
+        assert isinstance(order, Order), type(order)
+        assert isinstance(message_id, MessageId), type(message_id)
+
+        if order.is_ask():
+            return Ask(message_id, order.order_id, order.price, order.total_quantity - order.traded_quantity,
+                       order.timeout, order.timestamp)
+        else:
+            return Bid(message_id, order.order_id, order.price, order.total_quantity - order.traded_quantity,
+                       order.timeout, order.timestamp)
+
+    @property
+    def order_id(self):
+        """
+        :rtype: OrderId
+        """
+        return self._order_id
+
+    @property
+    def price(self):
+        """
+        :rtype: Price
+        """
+        return self._price
+
+    @property
+    def quantity(self):
+        """
+        :rtype: Quantity
+        """
+        return self._quantity
+
+    @quantity.setter
+    def quantity(self, quantity):
+        """
+        :param quantity: The new quantity
+        :type quantity: Quantity
+        """
+        assert isinstance(quantity, Quantity), type(quantity)
+        self._quantity = quantity
+
+    @property
+    def timeout(self):
+        """
+        Return when the tick is going to expire
+        :rtype: Timeout
+        """
+        return self._timeout
+
+    def is_ask(self):
+        """
+        :return: True if this tick is an ask, False otherwise
+        :rtype: bool
+        """
+        return self._is_ask
+
+    def is_valid(self):
+        """
+        :return: True if valid, False otherwise
+        :rtype: bool
+        """
+        time_tol = 10  # A small tolerance for the timestamp, to account for network delays
```

Comment:
Could you define this as a module variable at the top of the class, right now this is a magic constant in the middle of a class.